quodlibet-full fails to install on nixos-unstable

## Issue description

`quodlibet-full` fails to install on `nixos-unstable`. It worked before the recent update. It fails early on in the check phase.

```
running install tests
============================= test session starts ==============================
platform linux -- Python 3.7.2, pytest-3.9.3, py-1.7.0, pluggy-0.8.0
rootdir: /build/quodlibet-4.2.1, inifile: setup.cfg
plugins: xdist-1.24.1, forked-0.2
collected 3035 items / 1 skipped

tests/test___init__.py .....                                             [  0%]
tests/test_appdata_files.py ssss                                         [  0%]
tests/test_browsers___init__.py .........                                [  0%]
tests/test_browsers__base.py ........................................... [  2%]
.......................................................
builder for '/nix/store/rxl4kyd2frj3an3ywwzgmf8v9hg9p1mb-quodlibet-full-4.2.1.drv' failed with exit code 139
```

I'm guessing the fact that I have to build it from source at all is because it also fails everywhere else, so there's no cached build to download.

### Steps to reproduce

Try to install `quodlibet-full` on `nixos-unstable`

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.13, NixOS, 19.03.git.626233e (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `"nixos-19.03pre166308.626233eee6e"`
 - nixpkgs: `/pkgs/nixpkgs`

I get quite a bit further, attempting to build `quodlibet` rather than `quodlibet-full`.

The entire test suite finishes, then reports:

```
=================================== FAILURES ===================================
__________________________ TIconTheme.test_icon_theme __________________________

self = <tests.test_icons.TIconTheme testMethod=test_icon_theme>

    def test_icon_theme(self):
        theme = Gtk.IconTheme.get_default()
        theme.append_search_path(quodlibet.get_image_dir())

        for i in [
            "io.github.quodlibet.QuodLibet",
            "exfalso",
            "quodlibet-missing-cover"
        ]:
>           self.failUnless(theme.has_icon(i))
E           AssertionError: False is not true

tests/test_icons.py:25: AssertionError
_______________________ TTrayIcon.test_get_paused_pixbuf _______________________

self = <tests.plugin.test_trayicon.TTrayIcon testMethod=test_get_paused_pixbuf>

    def test_get_paused_pixbuf(self):
        get_paused_pixbuf = \
            self.modules["Tray Icon"].systemtray.get_paused_pixbuf

>       self.assertTrue(get_paused_pixbuf((1, 1), 0))
E       AssertionError: None is not true

tests/plugin/test_trayicon.py:54: AssertionError
____________________ TTrayIcon.test_new_with_paused_emblem _____________________

self = <tests.plugin.test_trayicon.TTrayIcon testMethod=test_new_with_paused_emblem>

    def test_new_with_paused_emblem(self):
        new_with_paused_emblem = \
            self.modules["Tray Icon"].systemtray.new_with_paused_emblem

        # too small source pixbuf
        for w, h in [(150, 1), (1, 150), (1, 1)]:
            pb = GdkPixbuf.Pixbuf.new(GdkPixbuf.Colorspace.RGB, True, 8, w, h)
            success, new = new_with_paused_emblem(pb)
            self.assertFalse(success)
            self.assertTrue(new)

        # those should work
        for w, h in [(20, 20), (10, 10), (5, 5), (150, 5), (5, 150)]:
            pb = GdkPixbuf.Pixbuf.new(GdkPixbuf.Colorspace.RGB, True, 8, w, h)
            success, new = new_with_paused_emblem(pb)
>           self.assertTrue(success)
E           AssertionError: False is not true

tests/plugin/test_trayicon.py:73: AssertionError
=============================== warnings summary ===============================
quodlibet/util/collection.py:23
  /build/quodlibet-4.2.1/quodlibet/util/collection.py:23: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Iterable

quodlibet/player/gstbe/util.py:139
  /build/quodlibet-4.2.1/quodlibet/player/gstbe/util.py:139: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    class TagListWrapper(collections.Mapping):

tests/test_browsers__base.py::new_test::test_active_filter
  /build/quodlibet-4.2.1/quodlibet/qltk/views.py:54: DeprecationWarning: Gtk.Misc.set_alignment is deprecated
    self.__clabel.set_alignment(0, 0.5)
  /build/quodlibet-4.2.1/quodlibet/qltk/songlist.py:401: DeprecationWarning: Gtk.TreeView.set_rules_hint is deprecated
    self.set_rules_hint(True)
  /build/quodlibet-4.2.1/quodlibet/qltk/x.py:283: DeprecationWarning: Gtk.ImageMenuItem.new_with_mnemonic is deprecated
    item = Gtk.ImageMenuItem.new_with_mnemonic(label)
  /build/quodlibet-4.2.1/quodlibet/qltk/x.py:284: DeprecationWarning: Gtk.ImageMenuItem.set_always_show_image is deprecated
    item.set_always_show_image(True)
  /build/quodlibet-4.2.1/quodlibet/qltk/x.py:287: DeprecationWarning: Gtk.ImageMenuItem.set_image is deprecated
    item.set_image(image)
  /build/quodlibet-4.2.1/quodlibet/qltk/menubutton.py:28: DeprecationWarning: Gtk.Arrow.new is deprecated
    Gtk.Arrow.new(arrow_type, Gtk.ShadowType.IN),
  /build/quodlibet-4.2.1/quodlibet/qltk/quodlibetwindow.py:884: DeprecationWarning: Gtk.ActionGroup.new is deprecated
    ag = Gtk.ActionGroup.new('QuodLibetWindowActions')
  /build/quodlibet-4.2.1/quodlibet/qltk/quodlibetwindow.py:877: DeprecationWarning: Gtk.ActionGroup.add_action is deprecated
    ag.add_action(Action(name="File", label=_("_File")))
  /build/quodlibet-4.2.1/quodlibet/qltk/quodlibetwindow.py:874: DeprecationWarning: Gtk.ActionGroup.add_action_with_accel is deprecated
    ag.add_action_with_accel(act, "<Primary>J")
  /build/quodlibet-4.2.1/quodlibet/qltk/quodlibetwindow.py:1009: DeprecationWarning: Gtk.RadioAction.join_group is deprecated
    act.join_group(first_action)
  /nix/store/x69n7z7m2gsl9wqm5zf2dixlhprq6xp8-python3.7-pygobject-3.30.4/lib/python3.7/site-packages/gi/overrides/Gtk.py:420: DeprecationWarning: Gtk.UIManager.insert_action_group is deprecated
    return Gtk.UIManager.insert_action_group(self, buffer, length)
  /nix/store/x69n7z7m2gsl9wqm5zf2dixlhprq6xp8-python3.7-pygobject-3.30.4/lib/python3.7/site-packages/gi/overrides/Gtk.py:417: DeprecationWarning: Gtk.UIManager.add_ui_from_string is deprecated
    return Gtk.UIManager.add_ui_from_string(self, buffer, length)
  /build/quodlibet-4.2.1/quodlibet/qltk/browser.py:207: DeprecationWarning: Gtk.UIManager.get_widget is deprecated
    return self._ui.get_widget(path)
  /build/quodlibet-4.2.1/quodlibet/qltk/quodlibetwindow.py:572: DeprecationWarning: Gtk.UIManager.get_accel_group is deprecated
    accel_group = ui.get_accel_group()
  /build/quodlibet-4.2.1/quodlibet/qltk/controls.py:217: DeprecationWarning: Gtk.Table.set_row_spacings is deprecated
    upper.set_row_spacings(3)
  /build/quodlibet-4.2.1/quodlibet/qltk/controls.py:218: DeprecationWarning: Gtk.Table.set_col_spacings is deprecated
    upper.set_col_spacings(3)
  /nix/store/x69n7z7m2gsl9wqm5zf2dixlhprq6xp8-python3.7-pygobject-3.30.4/lib/python3.7/site-packages/gi/overrides/Gtk.py:1523: DeprecationWarning: Gtk.Table.attach is deprecated
    Gtk.Table.attach(self, child, left_attach, right_attach, top_attach, bottom_attach, xoptions, yoptions, xpadding, ypadding)
  /build/quodlibet-4.2.1/quodlibet/qltk/views.py:1242: DeprecationWarning: Gtk.Misc.set_padding is deprecated
    label.set_padding(1, 1)

tests/test_browsers__base.py::new_test::test_filter_text
  /build/quodlibet-4.2.1/quodlibet/qltk/entry.py:270: DeprecationWarning: Gtk.Widget.override_color is deprecated
    self.override_color(Gtk.StateType.NORMAL, None)

tests/test_browsers__base.py::new_test::test_menu
  /build/quodlibet-4.2.1/quodlibet/browsers/playlists/menu.py:30: DeprecationWarning: Gtk.Widget.size_request is deprecated
    self.set_size_request(int(i.size_request().width * 2), -1)

tests/test_browsers__base.py::TBCoverGrid::test_active_filter
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_filter_albums
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_filter_other
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_filter_text
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_filters_caps
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_get_filter_text
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_init
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_key
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_menu
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_msic
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_name
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_pack_noshow_songpane
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_pack_unpack
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers__base.py::TBCoverGrid::test_save_restore
  /build/quodlibet-4.2.1/quodlibet/browsers/covergrid/main.py:226: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self.view = view = IconView(model_filter)

tests/test_browsers_playlists.py::TPlaylistsBrowser::test_deletion
  /build/quodlibet-4.2.1/quodlibet/qltk/__init__.py:329: DeprecationWarning: Gdk.Keymap.get_default is deprecated
    keymap = Gdk.Keymap.get_default()

tests/test_browsers_soundcloud.py::TestExtract::test_extract_complex
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:73: DeprecationWarning: Please use assertEqual instead.
    % (term, expected, terms))

tests/test_browsers_soundcloud.py::TestExtract::test_extract_composite_text
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:73: DeprecationWarning: Please use assertEqual instead.
    % (term, expected, terms))
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:73: DeprecationWarning: Please use assertEqual instead.
    % (term, expected, terms))

tests/test_browsers_soundcloud.py::TestExtract::test_extract_date
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:50: DeprecationWarning: Please use assertEqual instead.
    convert_time(now - 86400))

tests/test_browsers_soundcloud.py::TestExtract::test_extract_single_tag
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:73: DeprecationWarning: Please use assertEqual instead.
    % (term, expected, terms))

tests/test_browsers_soundcloud.py::TestExtract::test_extract_tag_inter
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:73: DeprecationWarning: Please use assertEqual instead.
    % (term, expected, terms))

tests/test_browsers_soundcloud.py::TestExtract::test_extract_tag_union
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:73: DeprecationWarning: Please use assertEqual instead.
    % (term, expected, terms))

tests/test_browsers_soundcloud.py::TestExtract::test_extract_unsupported
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:37: DeprecationWarning: Please use assertEqual instead.
    QueryType.INVALID)

tests/test_browsers_soundcloud.py::TestExtract::test_numeric_relative
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:73: DeprecationWarning: Please use assertEqual instead.
    % (term, expected, terms))
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:73: DeprecationWarning: Please use assertEqual instead.
    % (term, expected, terms))
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:73: DeprecationWarning: Please use assertEqual instead.
    % (term, expected, terms))
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:73: DeprecationWarning: Please use assertEqual instead.
    % (term, expected, terms))

tests/test_browsers_soundcloud.py::TestExtract::test_numeric_simple
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:73: DeprecationWarning: Please use assertEqual instead.
    % (term, expected, terms))

tests/test_browsers_soundcloud.py::TestHttpsDefault::test_setup_default
  /build/quodlibet-4.2.1/tests/test_browsers_soundcloud.py:85: DeprecationWarning: Please use assertTrue instead.
    msg="API client should use HTTPS")

tests/test_commands.py::TCommands::test_misc
  /build/quodlibet-4.2.1/quodlibet/qltk/quodlibetwindow.py:173: DeprecationWarning: Gtk.ToggleAction.set_active is deprecated
    self._stop_after.set_active(value)

tests/test_commands.py::TCommands::test_player
  /build/quodlibet-4.2.1/quodlibet/qltk/quodlibetwindow.py:1157: DeprecationWarning: Gtk.ToggleAction.get_active is deprecated
    if self.stop_after.get_active():

tests/test_commands.py::TCommands::test_set_browser
  /build/quodlibet-4.2.1/quodlibet/qltk/quodlibetwindow.py:1076: DeprecationWarning: Gtk.RadioAction.get_group is deprecated
    for action in self._browser_action.get_group():
  /build/quodlibet-4.2.1/quodlibet/qltk/quodlibetwindow.py:1077: DeprecationWarning: Gtk.Action.get_name is deprecated
    if action.get_name() == action_name:
  /build/quodlibet-4.2.1/quodlibet/qltk/quodlibetwindow.py:1019: DeprecationWarning: Gtk.RadioAction.get_current_value is deprecated
    browsers.get(current_action.get_current_value()))

tests/test_config.py::Tconfig::test_init_garbage_file
  /build/quodlibet-4.2.1/quodlibet/util/config.py:381: DeprecationWarning: This method will be removed in future versions.  Use 'parser.read_file()' instead.
    self._config.readfp(fileobj, filename)

tests/test_player_gst.py::TGstPlayerPrefs::test_main
  /build/quodlibet-4.2.1/quodlibet/player/gstbe/prefs.py:52: DeprecationWarning: Gtk.HScale.new is deprecated
    Gtk.Adjustment(value=duration, lower=0.2, upper=10))
  /build/quodlibet-4.2.1/quodlibet/player/gstbe/prefs.py:69: DeprecationWarning: Gtk.Button.set_alignment is deprecated
    gapless_button.set_alignment(0.0, 0.5)

tests/test_qltk___init__.py::TQltk::test_popup_menu_under_widget
  /nix/store/x69n7z7m2gsl9wqm5zf2dixlhprq6xp8-python3.7-pygobject-3.30.4/lib/python3.7/site-packages/gi/overrides/Gtk.py:1622: DeprecationWarning: Gtk.Menu.popup_for_device is deprecated
    self.popup_for_device(None, parent_menu_shell, parent_menu_item, func, data, button, activate_time)
  /build/quodlibet-4.2.1/quodlibet/qltk/__init__.py:227: DeprecationWarning: Gdk.Screen.get_width is deprecated
    screen_width = screen.get_width()
  /build/quodlibet-4.2.1/quodlibet/qltk/__init__.py:228: DeprecationWarning: Gdk.Screen.get_height is deprecated
    screen_height = screen.get_height()

tests/test_qltk___init__.py::TQltk::test_show_uri_with_existing_window
  /build/quodlibet-4.2.1/quodlibet/qltk/pluginwin.py:405: DeprecationWarning: Gtk.Button.set_focus_on_click is deprecated
    errors.set_focus_on_click(False)

tests/test_qltk_bookmarks.py::TBookmarks::test_menu_items
  /build/quodlibet-4.2.1/quodlibet/qltk/bookmarks.py:45: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    text = Gtk.Label(mark)
  /build/quodlibet-4.2.1/quodlibet/qltk/bookmarks.py:45: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    text = Gtk.Label(mark)

tests/test_qltk_cbes.py::TStandaloneEditor::test_constructor
  /build/quodlibet-4.2.1/quodlibet/qltk/cbes.py:38: DeprecationWarning: Gtk.Table.set_col_spacing is deprecated
    t.set_col_spacing(0, 3)

tests/test_qltk_cover.py::TCoverImage::test_big_window
  /build/quodlibet-4.2.1/quodlibet/qltk/cover.py:104: DeprecationWarning: Gdk.Screen.width is deprecated
    width = int(Gdk.Screen.width() / 1.8)
  /build/quodlibet-4.2.1/quodlibet/qltk/cover.py:105: DeprecationWarning: Gdk.Screen.height is deprecated
    height = int(Gdk.Screen.height() / 1.8)

tests/test_qltk_delete.py::TDeleteDialog::test_delete_files
  /build/quodlibet-4.2.1/quodlibet/qltk/delete.py:46: DeprecationWarning: Gtk.ScrolledWindow.add_with_viewport is deprecated
    win.add_with_viewport(Align(lab, border=6))

tests/test_qltk_exfalso.py::TExFalsoWindow::test_nothing
  /build/quodlibet-4.2.1/quodlibet/qltk/filesel.py:641: DeprecationWarning: Gtk.TreeView.get_vadjustment is deprecated
    self._saved_scroll_pos = filelist.get_vadjustment().get_value()

tests/test_qltk_information.py::TInformation::test_album
  /build/quodlibet-4.2.1/quodlibet/qltk/information.py:95: DeprecationWarning: Gtk.Table.set_homogeneous is deprecated
    t.set_homogeneous(False)

tests/test_qltk_prefs.py::TPreferencesWindow::test_ctr
  /build/quodlibet-4.2.1/quodlibet/qltk/prefs.py:75: DeprecationWarning: Gtk.Table.new is deprecated
    table = Gtk.Table.new(3, 3, True)

tests/test_util_config.py::TConfig::test_read_garbage_file
  /build/quodlibet-4.2.1/quodlibet/util/config.py:381: DeprecationWarning: This method will be removed in future versions.  Use 'parser.read_file()' instead.
    self._config.readfp(fileobj, filename)

tests/test_util_config.py::TConfig::test_versioning
  /build/quodlibet-4.2.1/quodlibet/util/config.py:381: DeprecationWarning: This method will be removed in future versions.  Use 'parser.read_file()' instead.
    self._config.readfp(fileobj, filename)
  /build/quodlibet-4.2.1/quodlibet/util/config.py:381: DeprecationWarning: This method will be removed in future versions.  Use 'parser.read_file()' instead.
    self._config.readfp(fileobj, filename)

tests/test_util_config.py::TConfig::test_versioning_disabled
  /build/quodlibet-4.2.1/quodlibet/util/config.py:381: DeprecationWarning: This method will be removed in future versions.  Use 'parser.read_file()' instead.
    self._config.readfp(fileobj, filename)

tests/test_util_config.py::TConfig::test_versioning_upgrade_func
  /build/quodlibet-4.2.1/quodlibet/util/config.py:381: DeprecationWarning: This method will be removed in future versions.  Use 'parser.read_file()' instead.
    self._config.readfp(fileobj, filename)

tests/plugin/test_albumart.py::TAlbumArt::testAlbumArtWindow
  /build/quodlibet-4.2.1/quodlibet/ext/songsmenu/albumart.py:397: DeprecationWarning: Gtk.Table.set_row_spacing is deprecated
    table.set_row_spacing(0, 5)
  /build/quodlibet-4.2.1/quodlibet/ext/songsmenu/albumart.py:699: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    search_labelraw = Gtk.Label('raw')
  /build/quodlibet-4.2.1/quodlibet/ext/songsmenu/albumart.py:704: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    search_labelclean = Gtk.Label('clean')
  /build/quodlibet-4.2.1/quodlibet/ext/songsmenu/albumart.py:725: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    search_labelresultsmax = Gtk.Label('limit')
  /build/quodlibet-4.2.1/quodlibet/ext/songsmenu/albumart.py:732: PyGTKDeprecationWarning: The keyword(s) "page_incr, step_incr" have been deprecated in favor of "page_increment, step_increment" respectively. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    page_incr=0, page_size=0)
  /build/quodlibet-4.2.1/quodlibet/ext/songsmenu/albumart.py:741: DeprecationWarning: Gtk.Alignment.set is deprecated
    search_button_box.set(1, 0, 0, 0)
  /build/quodlibet-4.2.1/quodlibet/ext/songsmenu/albumart.py:744: PyGTKDeprecationWarning: The keyword(s) "columns, rows" have been deprecated in favor of "n_columns, n_rows" respectively. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    search_table = Gtk.Table(rows=3, columns=4, homogeneous=False)

tests/plugin/test_console.py::TConsole::test_sidebar_plugin
  /build/quodlibet-4.2.1/quodlibet/ext/songsmenu/console.py:129: DeprecationWarning: Gtk.Widget.modify_font is deprecated
    self.view.modify_font(Pango.font_description_from_string('Monospace'))

tests/plugin/test_cover_download.py::TAlbumArt::test_cover_art_window
  /build/quodlibet-4.2.1/quodlibet/ext/songsmenu/cover_download.py:268: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    label = Gtk.Label(_("Preview size"))
  /build/quodlibet-4.2.1/quodlibet/ext/songsmenu/cover_download.py:277: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    label = Gtk.Label(_("Save destination"))

tests/plugin/test_duplicates_browser.py::TDuplicates::test_starts_up
  /build/quodlibet-4.2.1/quodlibet/qltk/views.py:625: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "model" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    super(BaseView, self).__init__(*args, **kwargs)
  /build/quodlibet-4.2.1/quodlibet/ext/songsmenu/duplicates.py:341: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    expand = Gtk.Button(_("Collapse / Expand all"))

tests/plugin/test_mpdserver.py::TMPDCommands::test_commands
  /build/quodlibet-4.2.1/quodlibet/qltk/quodlibetwindow.py:1138: DeprecationWarning: Gtk.ImageMenuItem.get_image is deprecated
    image = menu.get_image()

tests/plugin/test_prefs.py::TPrefs::test_all
  /build/quodlibet-4.2.1/quodlibet/ext/songsmenu/wikipedia.py:77: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    button = Gtk.Button(_("Edit Tags"))
  /build/quodlibet-4.2.1/quodlibet/ext/playorder/shufflebygrouping.py:138: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    label = Gtk.Label(label_text, selectable=True)
  /build/quodlibet-4.2.1/quodlibet/ext/playorder/shufflebygrouping.py:138: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    label = Gtk.Label(label_text, selectable=True)
  /build/quodlibet-4.2.1/quodlibet/ext/playorder/shufflebygrouping.py:138: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    label = Gtk.Label(label_text, selectable=True)
  /build/quodlibet-4.2.1/quodlibet/ext/playorder/shufflebygrouping.py:169: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "n_rows, n_columns" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    table = Gtk.Table(3, 2)
  /build/quodlibet-4.2.1/quodlibet/ext/playorder/shufflebygrouping.py:183: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    defaults = Gtk.Button(_("Reset to defaults"))
  /build/quodlibet-4.2.1/quodlibet/ext/events/waveformseekbar.py:707: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "value, lower, upper, step_increment, page_increment, page_size" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    0, 60000, 1000, 1000, 0)
  /build/quodlibet-4.2.1/quodlibet/ext/events/animosd/prefs.py:137: DeprecationWarning: Gdk.Screen.get_n_monitors is deprecated
    monitor_cnt = Gdk.Screen.get_default().get_n_monitors()
  /build/quodlibet-4.2.1/quodlibet/ext/events/animosd/prefs.py:207: DeprecationWarning: Gtk.FontButton.set_font_name is deprecated
    font.set_font_name(self.Conf.font)
  /build/quodlibet-4.2.1/quodlibet/ext/events/animosd/prefs.py:243: DeprecationWarning: Gtk.ColorButton.set_use_alpha is deprecated
    b.set_use_alpha(True)
  /build/quodlibet-4.2.1/quodlibet/ext/events/animosd/prefs.py:244: DeprecationWarning: Gtk.ColorButton.set_alpha is deprecated
    b.set_alpha(__floattocol(self.Conf.fill[3]))

tests/plugin/test_replaygain.py::TRGDialog::test_some_songs_needing_update
  /nix/store/x69n7z7m2gsl9wqm5zf2dixlhprq6xp8-python3.7-pygobject-3.30.4/lib/python3.7/site-packages/gi/overrides/Gtk.py:571: PyGTKDeprecationWarning: The keyword(s) "parent" have been deprecated in favor of "transient_for" respectively. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
    self._init(*args, **new_kwargs)

tests/plugin/test_trayicon.py::TTrayIcon::test_enable_disable
  /build/quodlibet-4.2.1/quodlibet/ext/events/trayicon/systemtray.py:180: DeprecationWarning: Gtk.StatusIcon.set_tooltip_markup is deprecated
    self._icon.set_tooltip_markup(util.escape(tooltip))
  /build/quodlibet-4.2.1/quodlibet/ext/events/trayicon/systemtray.py:160: DeprecationWarning: Gtk.StatusIcon.set_visible is deprecated
    self._icon.set_visible(False)

tests/plugin/test_trayicon.py::TTrayIcon::test_popup_menu
  /nix/store/x69n7z7m2gsl9wqm5zf2dixlhprq6xp8-python3.7-pygobject-3.30.4/lib/python3.7/site-packages/gi/overrides/Gtk.py:1622: DeprecationWarning: Gtk.StatusIcon.position_menu is deprecated
    self.popup_for_device(None, parent_menu_shell, parent_menu_item, func, data, button, activate_time)

tests/plugin/test_waveformseekbar.py::TWaveformSeekBar::test_main
  /build/quodlibet-4.2.1/quodlibet/ext/events/waveformseekbar.py:324: DeprecationWarning: Gtk.Widget.override_background_color is deprecated
    Gtk.StateFlags.NORMAL, Gdk.RGBA(alpha=0))
  /build/quodlibet-4.2.1/quodlibet/ext/events/waveformseekbar.py:492: DeprecationWarning: Gtk.StyleContext.get_background_color is deprecated
    bg_color = context.get_background_color(context.get_state())

-- Docs: https://docs.pytest.org/en/latest/warnings.html
======= 3 failed, 2963 passed, 70 skipped, 107 warnings in 68.97 seconds =======
builder for '/nix/store/zx76j9vclarm1qjpzs0qc2mc3sc0rkzc-quodlibet-4.2.1.drv' failed with exit code 1
error: build of '/nix/store/zx76j9vclarm1qjpzs0qc2mc3sc0rkzc-quodlibet-4.2.1.drv' failed
```

--- a/pkgs/applications/audio/quodlibet/default.nix
+++ b/pkgs/applications/audio/quodlibet/default.nix
@@ -33,13 +33,23 @@ python3.pkgs.buildPythonApplication rec {
 
   LC_ALL = "en_US.UTF-8";
 
+  pytestFlags = stdenv.lib.optionals (xineBackend || !withGstPlugins) [
+    "--ignore=tests/plugin/test_replaygain.py"
+  ] ++ [
+    # upstream does actually not enforce source code linting
+    "--ignore=tests/quality"
+    # build failure on Arch Linux
+    # https://github.com/NixOS/nixpkgs/pull/77796#issuecomment-575841355
+    "--ignore=tests/test_operon.py"
+  ];
+
   checkPhase = ''
     runHook preCheck
     env XDG_DATA_DIRS="$out/share:${gtk3}/share/gsettings-schemas/${gtk3.name}:$XDG_ICON_DIRS:$XDG_DATA_DIRS" \
       HOME=$(mktemp -d) \
       xvfb-run -s '-screen 0 800x600x24' dbus-run-session \
         --config-file=${dbus.daemon}/share/dbus-1/session.conf \
-        py.test ${stdenv.lib.optionalString (xineBackend || !withGstPlugins) " --ignore=tests/plugin/test_replaygain.py"} --ignore=tests/quality --ignore=tests/test_operon.py
+        py.test $pytestFlags
     runHook postCheck
   '';
 

--- a/pkgs/applications/audio/quodlibet/default.nix
+++ b/pkgs/applications/audio/quodlibet/default.nix
@@ -39,7 +39,7 @@ python3.pkgs.buildPythonApplication rec {
       HOME=$(mktemp -d) \
       xvfb-run -s '-screen 0 800x600x24' dbus-run-session \
         --config-file=${dbus.daemon}/share/dbus-1/session.conf \
-        py.test ${stdenv.lib.optionalString (xineBackend || !withGstPlugins) " --ignore=tests/plugin/test_replaygain.py"} --ignore=tests/quality
+        py.test ${stdenv.lib.optionalString (xineBackend || !withGstPlugins) " --ignore=tests/plugin/test_replaygain.py"} --ignore=tests/quality --ignore=tests/test_operon.py
     runHook postCheck
   '';
 

--- a/pkgs/applications/audio/quodlibet/default.nix
+++ b/pkgs/applications/audio/quodlibet/default.nix
@@ -18,7 +18,7 @@ python3.pkgs.buildPythonApplication rec {
 
   nativeBuildInputs = [ wrapGAppsHook gettext ];
 
-  checkInputs = with python3.pkgs; [ pytest pytest_xdist pyflakes pycodestyle polib xvfb_run dbus.daemon glibcLocales ];
+  checkInputs = [ gdk-pixbuf ] ++ (with python3.pkgs; [ pytest pytest_xdist polib xvfb_run dbus.daemon glibcLocales ]);
 
   buildInputs = [ gnome3.adwaita-icon-theme libsoup glib glib-networking gtk3 webkitgtk gdk-pixbuf keybinder3 gtksourceview libmodplug libappindicator-gtk3 kakasi gobject-introspection ]
     ++ (if xineBackend then [ xineLib ] else with gst_all_1;
@@ -35,11 +35,11 @@ python3.pkgs.buildPythonApplication rec {
 
   checkPhase = ''
     runHook preCheck
-    env XDG_DATA_DIRS="$out/share:${gtk3}/share/gsettings-schemas/${gtk3.name}:$XDG_DATA_DIRS" \
+    env XDG_DATA_DIRS="$out/share:${gtk3}/share/gsettings-schemas/${gtk3.name}:$XDG_ICON_DIRS:$XDG_DATA_DIRS" \
       HOME=$(mktemp -d) \
       xvfb-run -s '-screen 0 800x600x24' dbus-run-session \
         --config-file=${dbus.daemon}/share/dbus-1/session.conf \
-        py.test${stdenv.lib.optionalString (xineBackend || !withGstPlugins) " --ignore=tests/plugin/test_replaygain.py"}
+        py.test ${stdenv.lib.optionalString (xineBackend || !withGstPlugins) " --ignore=tests/plugin/test_replaygain.py"} --ignore=tests/quality
     runHook postCheck
   '';
 
@@ -65,6 +65,5 @@ python3.pkgs.buildPythonApplication rec {
 
     maintainers = with maintainers; [ coroa sauyon ];
     homepage = https://quodlibet.readthedocs.io/en/latest/;
-    broken = true;
   };
 }

